File: packages/server/internal/model/feedback.go
package model

import "time"

// Feedback 问题反馈
// 地图/移动端/后台提交的设备或路口问题反馈，可关联工单
type Feedback struct {
	ID            uint      `json:"id" gorm:"primaryKey"`
	DeviceHwID    *uint32   `json:"device_hw_id" gorm:"index;comment:关联设备硬件ID(可空)"`
	Intersection  string    `json:"intersection" gorm:"size:128;comment:路口位置"`
	Title         string    `json:"title" gorm:"size:128;comment:反馈标题"`
	Content       string    `json:"content" gorm:"type:text;comment:反馈内容"`
	Reporter      string    `json:"reporter" gorm:"size:64;comment:反馈人"`
	Contact       string    `json:"contact" gorm:"size:64;comment:联系方式"`
	Status        string    `json:"status" gorm:"size:16;default:open;comment:状态(open/processing/resolved/closed)"`
	WorkOrderID   *uint     `json:"work_order_id" gorm:"comment:关联工单ID"`
	ImageURL      string    `json:"image_url" gorm:"size:512;comment:反馈图片URL"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// TableName 指定表名
func (Feedback) TableName() string {
	return "feedbacks"
}

// 反馈状态常量
const (
	FeedbackOpen        = "open"        // 待处理
	FeedbackProcessing  = "processing"  // 处理中
	FeedbackResolved    = "resolved"    // 已解决
	FeedbackClosed      = "closed"      // 已关闭
)

File: packages/server-java/src/main/java/com/tsloms/server/model/Feedback.java
// 意见反馈实体：对齐 Go 版 Feedback（表 feedbacks）。
package com.tsloms.server.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import java.time.Instant;

/** 市民/移动端/后台提交的设备路灯问题反馈，可关联生成工单。 */
@Entity
@Table(name = "feedbacks", indexes = {
        @Index(name = "idx_feedback_device", columnList = "device_hw_id"),
})
public class Feedback {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column(name = "device_hw_id", length = 64)
    public String deviceHwId;

    /** 路口位置。 */
    @Column(name = "intersection", length = 128)
    public String intersection;

    @Column(name = "title", nullable = false, length = 128)
    public String title;

    @Column(name = "content", columnDefinition = "text")
    public String content;

    @Column(name = "reporter", length = 64)
    public String reporter;

    @Column(name = "contact", length = 64)
    public String contact;

    @Column(name = "status", nullable = false, length = 16)
    public String status = "open";

    /** 关联生成的工单 ID。 */
    @Column(name = "work_order_id")
    public Long workOrderId;

    @Column(name = "image_url", length = 512)
    public String imageUrl;

    @Column(name = "created_at", nullable = false)
    public Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    public Instant updatedAt;

    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        if (createdAt == null) {
            createdAt = now;
        }
        if (updatedAt == null) {
            updatedAt = now;
        }
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}
